packages/flutter/test/widgets/routes_test.dart

```diff
@@ -780,13 +780,46 @@ void main() {
       // Pop page three while replacement push is ongoing.
       navigator.currentState.pop();
       await tester.pump();
-      expect(secondaryAnimationPageOne.parent, isA<TrainHoppingAnimation>());
-      final TrainHoppingAnimation trainHopper2 = secondaryAnimationPageOne.parent as TrainHoppingAnimation;
-      expect(trainHopper2.currentTrain, animationPageTwo.parent);
-      expect(trainHopper.currentTrain, isNull); // Has been disposed.
+      // It should jump to the other train because train hopping will never
+      // happen
+      expect(secondaryAnimationPageOne.parent, isA<CurvedAnimation>());
```

Comment:
@goderbauer you wrote this test. However, something is unclear what the expected behavior should be.
The last pop trigger a train hopping where pri-animation of top most route = 0.16 reverse and sec-animation of the route below =0.67 forward. They will never meet and sec-animation just hang there until everything is disposed. This is what it looks like in real world
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/47866232/71218134-44434a00-2275-11ea-903a-69f370464fd6.gif)

Reply:
This is what it looks like after this pr
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/47866232/71218400-3215db80-2276-11ea-8368-359320c5d183.gif)

Reply:
These animations both look broken.